app/models/listed_taxon.rb

```diff
@@ -964,4 +965,18 @@ def establishment_means_description
     I18n.t( "establishment_means_descriptions.#{ key }", default: default )
   end
 
+  def reindex_observations
+    return true if taxon_id.blank? || place_id.blank?
+    return true unless previous_changes[:establishment_means]
+    Observation.
```

Comment:
I think we'll want to tweak this a little. With the `.delay().elastic_index!` setup, this is bypassing some of the logic in `.elastic_index!` for efficient delayed indexing. As written, the scope query will be run and all observations will be fetched and all of them will be serialized in the delayed job. We could improve on that with `Observation.elastic_index!( scope: ..., delay: true )` which will first fetch just the IDs of the obs and create jobs for them and only put the IDs in the delayed job handler - essentially don't call `.delay()` explicitly.

Another potential issue is that this will be using DB querying for obs in places, which can be very slow even for relatively small places, and prohibitively slow for large places. So I think we need to use Elasticsearch for determining which obs are in this taxon and place. I also think we should leave that querying to the delayed job so the user doesn't need to wait for us to figure out which obs to index later. So maybe there's a method like this called `reindex_observations` and another called `reindex_obsdervations_later`. The latter can be the rails callback and it just queues up the other one that'll query ES and kick off the indexing job

Reply:
K, I think I've done what you described if you want to take another look.